components/TaskList/TaskList.tsx

```diff
@@ -0,0 +1,59 @@
+import { useCallback } from 'react';
+import s from '../../stylesheets/TaskList.module.scss';
+import { groupByTheme } from '../../lib/utils';
+import { Form } from '../../data/flexibleForms/forms.types';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+
+interface Props {
+  form: Form;
+  completedSteps: string[];
+}
+
+const TaskList = ({ form, completedSteps }: Props): React.ReactElement => {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const groupByThemeCb = useCallback(groupByTheme, [form]);
+  const themes = groupByThemeCb(form);
+
+  return (
+    <ol className={s.taskList}>
+      {themes.map((theme, i) => (
+        <li key={theme.name}>
+          <h2 className={s.section}>
+            <span className={s.sectionNumber}>{i + 1}.</span> {theme.name}
+          </h2>
+
+          <ul className={s.items}>
+            {theme.steps.map((step) => (
+              <li className={s.item} key={step.id}>
+                <span className={s.taskName}>
+                  <Link href={`/submissions/${id}/steps/${step.id}`}>
+                    <a className="lbh-link">{step.name}</a>
+                  </Link>
+                </span>
+
+                {completedSteps.includes(step.id) ? (
```

Comment:
Can there ever be a large number of steps?

This is going to be an O(N^2) operation looking up every step if it exists in completed steps. Maybe O(NxM) I guess.

If instead of a string[] for our completed steps we use a set<string> then we remove the inner loop.

Just a thought but if the steps/completed steps are never that large it won't matter.

Reply:
i actually used a set for building the completedSteps array here: https://github.com/LBHackney-IT/social-care-recording/blob/main/lib/helpers.ts#L37. perhaps we can use that in more places